x-pack/plugins/endpoint/public/embeddables/resolver/view/submenu.tsx

```diff
@@ -0,0 +1,188 @@
+    defaultMessage: 'There was an error retrieving related events.',
+  }),
+  relatedAlerts: {
+    title: i18n.translate('xpack.endpoint.resolver.relatedAlerts', {
+      defaultMessage: 'Related Alerts',
+    }),
+  },
+  relatedEvents: {
+    title: i18n.translate('xpack.endpoint.resolver.relatedEvents', {
+      defaultMessage: 'Events',
+    }),
+  },
+};
+
+interface ResolverSubmenuOption {
+  optionTitle: string;
+  action: () => unknown;
+  prefix?: number | JSX.Element;
+}
+
+export type ResolverSubmenuOptionList =
+  | ResolverSubmenuOption[]
+  | 'waitingForRelatedEventData'
+  | typeof subMenuAssets.initialMenuStatus;
+
+const OptionList = React.memo(
+  ({
+    subMenuOptions,
+    isLoading,
+  }: {
+    subMenuOptions: ResolverSubmenuOptionList;
+    isLoading: boolean;
+  }) => {
+    const selectableOptions =
+      typeof subMenuOptions !== 'object'
+        ? []
+        : subMenuOptions.map((opt: ResolverSubmenuOption): {
+            label: string;
+            prepend?: ReactNode;
+          } => {
+            return opt.prefix
+              ? {
+                  label: opt.optionTitle,
+                  prepend: <span>{opt.prefix}</span>,
+                }
+              : {
+                  label: opt.optionTitle,
+                  prepend: <span />,
+                };
+          });
+    const [options, setOptions] = useState(selectableOptions);
+    return useMemo(
+      () => (
+        <EuiSelectable
+          singleSelection={true}
+          options={options}
+          onChange={newOptions => {
+            setOptions(newOptions);
+          }}
+          listProps={{ showIcons: true, bordered: true }}
+          isLoading={isLoading}
+        >
+          {list => list}
+        </EuiSelectable>
+      ),
+      [isLoading, options]
+    );
+  }
+);
+
+/**
+ * A Submenu to be displayed in one of two forms:
+ *   1) Provided a collection of `optionsWithActions`: it will call `menuAction` then - if and when menuData becomes available - display each item with an optional prefix and call the supplied action for the options when that option is clicked.
+ *   2) Provided `optionsWithActions` is undefined, it will call the supplied `menuAction` when its host button is clicked.
+ */
+export const NodeSubMenu = styled(
+  React.memo(
+    ({
+      menuTitle,
+      menuAction,
+      optionsWithActions,
+      className,
+    }: { menuTitle: string; className?: string; menuAction: () => unknown } & (
+      | {
+          optionsWithActions:
+            | ResolverSubmenuOptionList
+            | typeof subMenuAssets.initialMenuStatus
+            | 'waitingForRelatedEventData';
+        }
+      | { optionsWithActions?: undefined }
+    )) => {
+      const [menuIsOpen, setMenuOpen] = useState(false);
+      const handleMenuOpenClick = useCallback(
+        (clickEvent: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
+          clickEvent.preventDefault();

```

Comment:
can you explain why the preventDefault and stopPropagation is needed? maybe add a comment in the code explaining why that's important to have.

Reply:
https://github.com/elastic/kibana/pull/64121/commits/f601477aaf7b9d3190997562aba05f04a854ee48